services/debug/api_test.go

```diff
@@ -0,0 +1,89 @@
+package debug
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	errShhPost = errors.New("shh post failed")
+)
+
+func TestDebugSuite(t *testing.T) {
+	suite.Run(t, new(DebugSuite))
+}
+
+type DebugSuite struct {
+	suite.Suite
+	p    *MockPoster
+	api  *PublicAPI
+	ctx  context.Context
+	msg  whisper.NewMessage
+	hash hexutil.Bytes
+}
+
+func (s *DebugSuite) SetupTest() {
+	ctrl := gomock.NewController(s.T())
+	s.p = NewMockPoster(ctrl)
+	w := whisper.New(nil)
+	service := New(w)
+	service.SetPoster(s.p)
+
+	symID, err := w.GenerateSymKey()
+	s.NoError(err)
+	s.msg = whisper.NewMessage{
+		SymKeyID:  symID,
+		PowTarget: whisper.DefaultMinimumPoW,
+		PowTime:   200,
+		Topic:     whisper.TopicType{0x01, 0x01, 0x01, 0x01},
+		Payload:   []byte("hello"),
+	}
+	s.hash = []byte("hash")
+
+	s.api = NewAPI(service)
+}
+
+func (s *DebugSuite) TestPostconfirmErrors() {
+	var testCases = []struct {
+		name                string
+		prepareExpectations func(*DebugSuite)
+		expectedError       error
+		expectedHash        hexutil.Bytes
+	}{
+		{
+			name: "post errored",
```

Comment:
both conditions can be tested without mocking
1. post will return error is neither sym or asym key is passed
2. timeout will fire if it will be shorter than ttl

Reply:
But, I will need a fully working `shhext` service injected, and thay're unit tests. I guess I could add tests with those conditions as part of e2e suite, but I think it doesn't make sense as they will be slower.

Reply:
Not sure I see the reason to avoid unit tests in this case, since it means we're testing the logic directly with the least amount of other code involved as possible.